src/sentry/api/endpoints/organization_discover.py

```diff
@@ -73,10 +81,26 @@ def __init__(self, *args, **kwargs):
         )
         self.arrayjoin = match if match else None
 
+        has_start = bool(data.get('start'))
+        has_end = bool(data.get('end'))
+        has_range = bool(data.get('range'))
+
+        if has_start != has_end or has_range == has_start:
+            raise serializers.ValidationError("Either start and end dates or range is required")
+
     def validate(self, data):
         data['arrayjoin'] = self.arrayjoin
+
         return data
 
+    def validate_range(self, attrs, source):
+        # Populate start and end if only range is provided
+        if (attrs.get(source)):
+            attrs['start'] = timezone.now() - parse_stats_period(attrs[source])
```

Comment:
You could have `timezone.now() - None` here. Will the resulting error create a 500, or a 4xx error letting the user know they goofed?

Do you need to cap the range? Or can users request `1000000d`?

Reply:
Fixed this to return a 403. We don't cap the range at all though (but there will obviously be missing data after ~90 days)